src/groupBy.js

```diff
@@ -48,7 +47,10 @@ var append = require('./append');
 module.exports = _curry2(_dispatchable('groupBy', _xgroupBy, function groupBy(fn, list) {
   return _reduce(function(acc, elt) {
     var key = fn(elt);
-    acc[key] = append(elt, acc[key] || (acc[key] = []));
+    if (acc[key] == null) {
```

Comment:
Actually, the existing code *happens* to do the right thing:

```javascript
> R.append(1, Object.prototype.toString)
[1]
```

Let's include an assertion such as `eq(R.groupBy(R.identity, ['toString']), {toString: ['toString']})`, and update the check to use `_has`.